Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41466&attempt_id=c96aed7f-22c0-4de7-a9bd-95eaca345231)

Attempted: haproxy

<details><summary>Partial log (click to expand)</summary><p>

```
include/proto/fd.h:154:31: error: use of undeclared identifier 'all_threads_mask'
        if ((fdtab[fd].update_mask & all_threads_mask) == 0) {
                                     ^
src/ev_poll.c:115:30: error: use of undeclared identifier 'all_threads_mask'
                        fdtab[fd].update_mask &= ~all_threads_mask;
                                                  ^
2 errors generated.
make: *** [Makefile:909: src/ev_poll.o] Error 1
builder for '/nix/store/6g9i8i7y2h1g4yfizmnla2riv57j66ad-haproxy-1.8.9.drv' failed with exit code 2
error: build of '/nix/store/6g9i8i7y2h1g4yfizmnla2riv57j66ad-haproxy-1.8.9.drv' failed
```
</p></details>

--- a/nixos/release.nix
+++ b/nixos/release.nix
@@ -300,6 +300,7 @@ in rec {
   tests.grafana = callTest tests/grafana.nix {};
   tests.graphite = callTest tests/graphite.nix {};
   tests.hardened = callTest tests/hardened.nix { };
+  tests.haproxy = callTest tests/haproxy.nix {};
   tests.hibernate = callTest tests/hibernate.nix {};
   tests.hitch = callTest tests/hitch {};
   tests.home-assistant = callTest tests/home-assistant.nix { };

--- a/nixos/tests/haproxy.nix
+++ b/nixos/tests/haproxy.nix
@@ -1,0 +1,41 @@
+import ./make-test.nix ({ pkgs, ...}: {
+  name = "haproxy";
+  nodes = {
+    machine = { config, ...}: {
+      imports = [ ../modules/profiles/minimal.nix ];
+      services.haproxy = {
+        enable = true;
+        config = ''
+          defaults
+            timeout connect 10s
+
+          backend http_server
+            mode http
+            server httpd [::1]:8000
+
+          frontend http
+            bind *:80
+            mode http
+            use_backend http_server
+        '';
+      };
+      services.httpd = {
+        enable = true;
+        documentRoot = pkgs.writeTextDir "index.txt" "We are all good!";
+        adminAddr = "notme@yourhost.local";
+        listen = [{
+          ip = "::1";
+          port = 8000;
+        }];
+      };
+    };
+  };
+  testScript = ''
+    startAll;
+    $machine->waitForUnit('multi-user.target');
+    $machine->waitForUnit('haproxy.service');
+    $machine->waitForUnit('httpd.service');
+    $machine->succeed('curl -k http://localhost:80/index.txt | grep "We are all good!"');
+
+  '';
+})

--- a/pkgs/tools/networking/haproxy/default.nix
+++ b/pkgs/tools/networking/haproxy/default.nix
@@ -23,7 +23,11 @@ stdenv.mkDerivation rec {
       url = "https://git.haproxy.org/?p=haproxy-1.8.git;a=patch;h=17514045e5d934dede62116216c1b016fe23dd06";
       sha256 = "0hzcvghg8qz45n3mrcgsjgvrvicvbvm52cc4hs5jbk1yb50qvls7";
     })
-  ];
+  ] ++ stdenv.lib.optional stdenv.isDarwin (fetchpatch {
+    name = "fix-darwin-no-threads-build.patch";
+    url = "https://git.haproxy.org/?p=haproxy-1.8.git;a=patch;h=fbf09c441a4e72c4a690bc7ef25d3374767fe5c5;hp=3157ef219c493f3b01192f1b809a086a5b119a1e";
+    sha256 = "16ckzb160anf7xih7mmqy59pfz8sdywmyblxnr7lz9xix3jwk55r";
+  });
 
   buildInputs = [ openssl zlib ]
     ++ stdenv.lib.optional useLua lua5_3

